### PR TITLE
Update packages and test Node 6, 8, and 10 in CI

### PR DESCRIPTION
**Pull request by nbrookie**
Give packages some ❤️ and also update which node environments we test against to include node 6, 8, and 10. Drop support for node 4.

Breaking changes:
- Update babel config to no longer support node 4
- Drop support for node 4 and 7 in CI
- Update all packages, including the following which now require node 6 or above:
  - broccoli-babel-transpiler
  - @babel/core
  - broccoli-test-helper
  - broccoli-persistent-filter
  - qunit
  - broccoli-merge-trees
  - fs-tree-diff
  - slash
  - node-watch
  - broccoli
  - amd-name-resolver
  - json5
  - @babel/parser
  - fs-updater
  - sane
  - execa
  - get-stream

### CODE REVIEW

**Comment by Turbo87** [comment]
@nbrookie thanks for the PR! 🙏 

can you summarize what breaking changes are included in this PR?

**Comment by nbrookie** [comment]
@Turbo87 Sure thing. Updated the description to include more details about the breaking changes.

**Review by kellyselden — APPROVED** [review]

**Review by rwjblue — APPROVED** [review]
